templates/preferences.html

```diff
@@ -0,0 +1,210 @@
+                    </td>
+                </tr>
+                </tbody>
+            </table>
+
+            {% if not session.is_admin %}
+                <button type="button" id="addPreferencesButton" class="btn btn-primary">Edit Preferences
+                </button>
+            {% endif %}
+            <button type="button" id="savePreferencesButton" class="btn btn-success"
+                    style="display: none;">
+                Save Preferences
+            </button>
+            <button type="button" id="cancelEditButton" class="btn btn-secondary"
+                    style="display: none;">
+                Cancel
+            </button>
+            <button type="button" id="addRowButton" class="btn btn-primary" style="display: none;">
+                Add Row
+            </button>
+        </form>
+    </div>
+{% endblock %}
+
+{% block additionalfooter %}
+    <script>
+        $(document).ready(function () {
+            // Handle year select change
+            handleYearSelectChange();
+
+            //Handle preferences
+            $('#addPreferencesButton').click(function () {
+                $('#staticPreferencesTable').hide();
+                $('#editablePreferencesTable').show();
+                $('#addPreferencesButton').hide();
+                $('#savePreferencesButton').show();
+                $('#cancelEditButton').show();
+                $('#addRowButton').show();
+            });
+
+            $('#cancelEditButton').click(function () {
+                $('#editablePreferencesTable').hide();
+                $('#staticPreferencesTable').show();
+                $('#addPreferencesButton').show();
+                $('#savePreferencesButton').hide();
+                $('#cancelEditButton').hide();
+                $('#addRowButton').hide();
+            });
+
+            $('#addRowButton').click(function () {
+                const newRow = $('#template-row').clone().removeAttr('id').removeAttr('style');
+                $('#editablePreferencesTable tbody').append(newRow);
+
+                $("#editablePreferencesTable tbody").sortable({
+                    items: 'tr',
+                    handle: '.move-icon',
+                    axis: 'y',
+                    cursor: 'move'
+                });
+            });
+
+            // Event listener pour le bouton de suppression
+            $(document).on('click', '.delete-icon', function () {
+                $(this).closest('tr').remove();
+            });
+
+            $('#savePreferencesButton').click(function () {
+                const preferences = [];
+                !$('#editablePreferencesTable tbody tr').not('#template-row').each(function () {
+                    const courseValue = $(this).find('select.course-select').val();
+
+                    if (courseValue) {
+                        const [course_id, course_year] = courseValue.split("-");
+                        preferences.push({course_id: course_id, course_year: course_year});
```

Comment:
Do we really need to pass the year as argument for each course ? Preferences are registered for one year at the time.

It seems we could end up, by modifying the form content, to edit past and future preferences, if the course is defined. Moreover, the `save_preference` route seems to rely on the `current_year` for part of its operations.